packages/core/test/event.test.ts

```diff
@@ -67,9 +69,14 @@ describe('SCXML events', () => {
       }
     });
 
-    const service = interpret(machine).start();
-
-    expect(service.state.context.childOrigin).toBe('callback_child');
+    interpret(machine)
```

Comment:
q: did anything change here that this test has to be asynchronous right now?

Reply:
Nope, just a refactoring to match how other services are "idiomatically" tested.

Although, we _should_ discourage reading `service.state` directly, and maybe disallow it in favor of `service.getSnapshot()`

Reply:
> Nope, just a refactoring to match how other services are "idiomatically" tested.

I agree that this might be "idiomatically" correct, however, I've found out that writing "non-idiomatic" tests (aka tests executed synchronously) can reveal more edge cases while writing those tests. It's because some problems only manifest themselves when we deal with synchronous execution, ping-ponging between different actors - this is true for XState, Redux-Saga, and even RxJS because those problems can be classified as scheduling problems and all of those libs deal with this and try to manage the complexity involved with communicating different "processes" etc